### PR TITLE
fix: 修复switcher组件布局问题

### LINKED ISSUE #1264: 🐛行列切换后，数据没了。官网例子。
### 🏷 Version

| Package      | Version |
| -------------- | --------- |
| @antv/s2       | 1.14.0         |
| @antv/s2-react | 1.13.0        |

### 🔗 Reproduce Link

访问：

https://s2.antv.vision/zh/examples/react-component/header#default

把【城市】从行头拖到列头，点击确定。然后数据就消失了

### LINKED ISSUE #1265: 🐛维度切换头部组件，当行/列没有字段时。 行/列编辑器会消失，没办法继续操作
### 🏷 Version

| Package      | Version |
| -------------- | --------- |
| @antv/s2       | 1.14.0         |
| @antv/s2-react | 1.13.0        |

### 🔗 Reproduce Link

官网例子，访问：

https://s2.antv.vision/zh/examples/react-component/switcher#pivot-header

把【列头】里面的type拖到行头里面去。 点击确定。

再次点击【行列切换】按钮，发现列头消失。 我无法再把type拖回列头

### PR DESCRIPTION
**Pull request by wjgogogo**
### 👀 PR includes

✨ Feature

- [ ] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [x] Solve the issue and close #1264 
- [x] Solve the issue and close #1265  

🔧 Chore

- [ ] Test case
- [x] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
|      |     |

### 🔍 Self Check before Merge

- [ ] Add or update relevant Docs.
- [ ] Add or update relevant Demos.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-core/__tests__/unit/sheet-type/pivot-sheet-spec.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // eslint-disable-next-line max-classes-per-file
 import { getContainer } from 'tests/util/helpers';
-import * as dataCfg from 'tests/data/simple-data.json';
```
Comment (wjgogogo) [inline packages/s2-core/__tests__/unit/sheet-type/pivot-sheet-spec.ts:3]: 这种导入方式，会多个`default`属性，而且`default`属性内容和`simple-data.json`内容是一样的

File: packages/s2-core/src/sheet-type/spread-sheet.ts [hunk 2]
```diff
@@ -324,8 +324,7 @@ export abstract class SpreadSheet extends EE {
    */
   public setDataCfg(dataCfg: S2DataConfig) {
     this.store.set('originalDataCfg', dataCfg);
-    const newDataCfg = clone(dataCfg);
-    this.dataCfg = getSafetyDataConfig(newDataCfg);
+    this.dataCfg = getSafetyDataConfig(this.dataCfg, dataCfg);
```
Comment (wjgogogo) [inline packages/s2-core/src/sheet-type/spread-sheet.ts:327]: dataCfg更新逻辑应该和options一样，采用`merge(old,new)`方式

**Review by lijinke666 — APPROVED** [review]

**Pull request merged**